raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -558,58 +576,44 @@ contract TokenNetwork is Utils {
             participant2_state
         );
 
-        participant1_amount = (
-            participant1_state.deposit +
+        // Calculate the netted transferred amounts
+        // We are sure this does not underflow
+        participant1_netted_transferred_amount = (
             participant2_transferred_amount -
-            participant1_state.withdrawn_amount -
             participant1_transferred_amount
         );
 
-        // There are 2 cases that require attention here:
-        // case1. If participant1 does NOT provide a balance proof or provides an old balance proof
-        // case2. If participant2 does NOT provide a balance proof or provides an old balance proof
-        // The issue is that we need to react differently in both cases. However, both cases have
-        // an end result of participant1_amount > total_available_deposit. Therefore:
-
-        // case1: participant2_transferred_amount can be [0, real_participant2_transferred_amount)
-        // This can trigger an underflow -> participant1_amount > total_available_deposit
-        // We need to make participant1_amount = 0 in this case, otherwise it can be
-        // an attack vector. participant1 must lose all/some of its tokens if it does not
-        // provide a valid balance proof.
-        if (
-            (participant1_state.deposit + participant2_transferred_amount) <
-            (participant1_state.withdrawn_amount + participant1_transferred_amount)
-        ) {
-            participant1_amount = 0;
-        }
+        // case2. participant1_transferred_amount can be [0, real_participant1_transferred_amount)
+        // participant1_netted_transferred_amount is bigger than expected and can cause an overflow.
+        // We just take the biggest amount that we can in this case.
+        participant1_amount = failsafe_addition(
+            participant1_netted_transferred_amount,
+            participant1_state.deposit
+        );
```

Comment:
We can potentially try to re-add the overflowed amount again after subtracting the withdrawn amount. Not entirely sure it is needed. Opinions?

Reply:
If we have valid balance proofs, `participant1_netted_transferred_amount + participant1_state.deposit` should never overflow, because we already ensure that `participant1_state.deposit + participant2_state.deposit` does not overflow.

If we do not have valid balance proofs, then it is fair to cap  `participant1_netted_transferred_amount + participant1_state.deposit` at `MAX_UINT256`.

So, I think the current approach is ok. We don't need to do what I proposed in my comment above.

Reply:
I think `participant1_state.deposit + participant2_state.deposit < MAX_UINT256 -> participant1_netted_transferred_amount + participant1_state.deposit < MAX_UINT256` does not hold, or do we have a require somewhere that I missed? 
I would, however, also do not add logic to deal with the case we're it might, as it is imo pretty unlikely to occur in a non-adverse scenario.

Reply:
> does not hold

why? 
I was talking about the case where balance proofs are computed valid. Therefore, if a participant needs to receive a **positive** netted amount, this amount can only come from the partner's `deposit` => `participant1_netted_transferred_amount` <= `participant2_state.deposit`

Reply:
participant1_netted_transferred_amount = 2**255 <= 2**255 = participant2_state.deposit would be a valid variable assignment for a balance proof, that would overflow, no?

Reply:
> So, I think the current approach is ok. We don't need to do what I proposed in my comment above.

Yes it is definitely fair to cap participant1 at UINT256_safe_max at the case where participant2 has cheated.

As a sidenote, the comments here are really nice and explain the edge cases nicely but we should also document those in the spec eventually